server/src/main/java/org/opensearch/repositories/RepositoriesService.java

```diff
@@ -457,18 +457,22 @@ public void applyClusterState(ClusterChangedEvent event) {
                             || previousMetadata.settings().equals(repositoryMetadata.settings()) == false) {
                             // Previous version is different from the version in settings
```

Comment:
It appears we can still delete a system repository currently. Is this being handled in the lock repository change @gbbafna?
https://github.com/opensearch-project/OpenSearch/blob/91f9c5dbf417e179c5ab00e49332c17c84661008/server/src/main/java/org/opensearch/repositories/RepositoriesService.java#L436-L446

Reply:
Discussed offline, this is being handled separately